tutor/src/screens/assignment-edit/tasking.js

```diff
@@ -91,15 +95,18 @@ class Tasking extends React.Component {
       }
 
       if(!didUserChangeDatesManually) {
+        // *dueAt* means that assignment creation was started from the calendar
+        // if due date is changed and assignment creation started from the calendar,
+        // then change dates according to grading template intervals
         if(dueAt) {
           if (gradingTemplate) {
             // this will also reset the due_at to template time
             t.onGradingTemplateUpdate(gradingTemplate, t.due_at, { dateWasManuallySet: true });
           }
-
           this.form.setFieldValue(`tasking_plans[${index}].opens_at`, t.opens_at);
           this.form.setFieldValue(`tasking_plans[${index}].closes_at`, t.closes_at);
         }
+        // otherwise do not enforce grading template dates interval anymore
         else
           this.props.ux.setDidUserChangeDatesManually(true);
```

Comment:
this looks like there's two consective if statements that both check `(!didUserChangeDatesManually)`. (lines 68 & 97) Should they be combined?

Reply:
they are on different methods. They do almost the same, but one checks for `!dueAt` and the other one `dueAt`, and also one sets the `due_at` and the other one the `opens_at`.